fix: an occasional error in betanet workflow

ShookLyngs opened the pull request:
There's an occasional error with Godwoken-Tests on betanet: when test case comparing `expected balance` with `current balance` and expect them to be equal, sometimes they're not equal to the other.

I think the reasons why it happens are:
- Developers use the same set of test accounts in the same network
- `get-balance` comparison is not a guaranteed way of check the true cost of a transaction

So the commits in the PR provides:
1. Use GitHub Actions secrets in workflow, so in the repo we can set a different set of accounts for workflow testing only
2. Replace most `get-balance` validations with `tx/receipt` validations
3. Other changes (changed`.gitignore`, fixed contract warning, etc)

ShookLyngs committed:
fix: replace get-balance validates with tx/receipt validates

ShookLyngs committed:
feat: use secrets in betanet testing

keroro520 commented on .github/workflows/betanet_v1.1.yml:
```diff
@@ -41,4 +41,6 @@ jobs:
-      run: npm install ; npm run test:gw_betanet_v1
+      run: |
+        npm install; 
+        PRIVATE_KEY=${{ secrets.GODWOKEN_PRIVATE_KEY }} PRIVATE_KEY2=${{ secrets.GODWOKEN_PRIVATE_KEY2 }} npm run test:gw_betanet_v1
```
1. It would be more meaningful if the secret name included the network name, `${{ secrets.GODWOKEN_BETANET_PRIVATE_KEY }}`
2. BTW, using [`jobs.<job_id>.env`](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idenv) is clearer.

keroro520 commented on .github/workflows/betanet_v1.1.yml:
```diff
@@ -41,4 +41,6 @@ jobs:
     - name: Run tests
       working-directory: contracts
-      run: npm install ; npm run test:gw_betanet_v1
+      run: |
```
```suggestion
      run: |
      if: ${{ secrets.GODWOKEN_BETANET_PRIVATE_KEY != '' }}
```

Fork repos may not set this secret. Skip them.

ShookLyngs replied:
> Fork repos may not set this secret. Skip them.

Should I just add the condition to the job, so if fork repo has no serect set, then skip the whole test.
```yml
jobs:
  contract-tests:
    if: ${{ secrets.GODWOKEN_TESTNET_PRIVATE_KEY != '' }} && ${{ secrets.GODWOKEN_TESTNET_PRIVATE_KEY2 != '' }}
```

ShookLyngs replied:
Codes in the last comment was a wrong attempt.
I have commited some new changes about the `secret checking` function, now it should work.

Flouse commented:
Please resolve the conflicts.

ShookLyngs committed:
Merge remote-tracking branch 'origin/develop' into develop

# Conflicts:
#	.github/workflows/betanet_v1.1.yml

ShookLyngs committed:
feat: add run condition to testnet(betanet) testing

ShookLyngs committed:
fix: condition to run testnet(betanet) testing

ShookLyngs commented:
Commited with some changes:
1. `betanet` is now called `testnet` (merged latest commits and changed some names)
2. Added `secrets check` before running tests
3. Set environment variables with [jobs.<job_id>.env](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idenv)
